sentry_sdk/sessions.py

```diff
@@ -19,32 +20,56 @@
     from typing import Union
 
 
-def is_auto_session_tracking_enabled(hub=None):
-    # type: (Optional[sentry_sdk.Hub]) -> Union[Any, bool, None]
-    """Utility function to find out if session tracking is enabled."""
-    # TODO: add deprecation warning
-
-    if hub is None:
-        hub = sentry_sdk.Hub.current
```

Comment:
Imo to deprecate the hub based session tracking we could add a deprecation warning here if `hub is not None` and explain that in the next major they can not use the `hub` as a parameter in the next major, but can pass a `scope` parameter.

When we do the next major we do what is written in the comment in sessions.py:
```
TODO: This (is_auto_session_tracking_enabled_scope) uses the new scopes. When the Hub is removed, the function is_auto_session_tracking_enabled should be removed and this function should be renamed to is_auto_session_tracking_enabled.
```

Reply:
Discussed offline. The `is_auto_session_tracking_enabled` deprecation is now in a separate PR (#3428). After clarifying the reasoning for introducing a new function was that it would allow us in our own code to avoid using deprecated APIs, @antonpirker agreed that introducing `track_sessions` was a good idea